tests/distributions/test_custom_distribution.py

```diff
@@ -0,0 +1,280 @@
+#   Copyright 2024 The PyMC Developers
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+import warnings
+
+import cloudpickle
+import numpy as np
+import numpy.random as npr
+import pytensor
+import pytensor.tensor as pt
+import pytest
+import scipy.stats as st
+
+import pymc as pm
```

Comment:
Ideally we don't import pymc as pm, but import whatever functions are needed instead

Reply:
Hi @ricardoV94, had a look at this.

What we did was just copy over the imports needed from `test_distribution.py` to `test_custom_distribution.py` without altering them.

It seems to me that the import was done in this way in the original `test_distribution.py` to keep, for example, `pm.Normal` distinct from `pymc.distributions.Normal` (imported in [L38 here](https://github.com/pymc-devs/pymc/blob/main/tests/distributions/test_distribution.py#L38)).

In this regard see for example [L107](https://github.com/pymc-devs/pymc/blob/main/tests/distributions/test_distribution.py#L38) (uses `pm.Normal`) vs [L173](https://github.com/pymc-devs/pymc/blob/main/tests/distributions/test_distribution.py#L173) (uses `pymc.distribution.Normal`).

I found a few other cases to which the above applies (`logp`, `Model`).

Thoughts?

Reply:
Yeah that's a lazy and discouraged pattern, sinxe we're moving thigs away it's a good time as any to clean it up and import things properly

Reply:
I addressed that after the rebase, thanks!